From 39efe4cf20b036ccc36cf8bf594f1bb5aab50085 Mon Sep 17 00:00:00 2001
From: Yuyao Lin <linyuyao1@huawei.com>
Date: Tue, 15 Nov 2022 09:28:55 +0800
Subject: [PATCH 0024/1398] [Huawei] Revert "posix-cpu-timers: Make timespec to
 nsec conversion safe"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186339

--------------------------------

This reverts commit 098b0e01a91c42aaaf0425605cd126b03fcb0bcf.

Function timespec64_to_ns() Add the upper and lower limits check in
commit cb47755725da ("time: Prevent undefined behaviour in timespec64_to_ns()"),
timespec64_to_ktime() only check the upper limits, so revert this
patch can fix overflow.

Signed-off-by: Yuyao Lin <linyuyao1@huawei.com>
---
 kernel/time/posix-cpu-timers.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/kernel/time/posix-cpu-timers.c b/kernel/time/posix-cpu-timers.c
index d6b46346be1f..c97bc3e3b210 100644
--- a/kernel/time/posix-cpu-timers.c
+++ b/kernel/time/posix-cpu-timers.c
@@ -591,11 +591,7 @@ static int posix_cpu_timer_set(struct k_itimer *timer, int timer_flags,
 		return -ESRCH;
 	}
 
-	/*
-	 * Use the to_ktime conversion because that clamps the maximum
-	 * value to KTIME_MAX and avoid multiplication overflows.
-	 */
-	new_expires = ktime_to_ns(timespec64_to_ktime(new->it_value));
+	new_expires = timespec64_to_ns(&new->it_value);
 
 	/*
 	 * Protect against sighand release/switch in exit/exec and p->cpu_timers
-- 
2.17.1

